### PR TITLE
Run mypy's stubtest too

### PR DESCRIPTION
**Pull request by altendky**
Note:
- This is using an unreleased mypy version for the fix in https://github.com/python/mypy/commit/538d36481526135c44b90383663eaa177cfc32e3

Draft for:
- [x] ~~Fixing~~ Addressing the reported issues
- [ ] ~~Use released mypy version with proper constraint~~
  - If anyone else wants to hold this up on the release, I understand, but I don't feel the need to anymore since it is a test-only dependency.
- [x] Deal with (pile of) ignored items
  - `stubtest.allowed.to_review` is to be addressed in #106

### CODE REVIEW

**Comment by altendky** [comment]
@BryceBeagle, here are the remaining to-be-reviewed allowances.  I'm not sure what to make of them yet.  They are all 'enumerations' and exactly the ones that have the enumerators nested inside the enumeration.  Though there are only 25.  So, maybe this is just details related to #106.

<details> <summary>Full stubtest output</summary>

```
error: PyQt5.QtBluetooth.QBluetoothDeviceInfo.Field.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtCore.QAbstractItemModel.CheckIndexOption.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtCore.QByteArray.Base64DecodingStatus.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtCore.QCalendar.System.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtCore.QCborKnownTags.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtCore.QCborSimpleType.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtCore.QDateTime.YearRange.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtCore.Qt.HighDpiScaleFactorRoundingPolicy.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtGui.QColorSpace.Primaries.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtGui.QColorSpace.TransferFunction.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtGui.QTextBlockFormat.MarkerType.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNetwork.QOcspCertificateStatus.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNetwork.QOcspRevocationReason.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNetwork.QSslCertificate.PatternSyntax.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNetworkAuth.QAbstractOAuth.ContentType.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNetworkAuth.QAbstractOAuth.Error.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNetworkAuth.QAbstractOAuth.Stage.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNetworkAuth.QAbstractOAuth.Status.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNetworkAuth.QOAuth1.SignatureMethod.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNetworkAuth.QOAuth1Signature.HttpRequestMethod.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtNfc.QNearFieldManager.AdapterState.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtTest.QAbstractItemModelTester.FailureReportingMode.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtWebEngineCore.QWebEngineUrlScheme.Syntax.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtWebEngineWidgets.QWebEnginePage.LifecycleState.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)

error: PyQt5.QtWidgets.QActionGroup.ExclusionPolicy.__new__ is inconsistent, stub does not have argument "value"
Stub: at line 100
def (cls: Type[builtins.object]) -> Any
Runtime: at line 557 in file /usr/lib/python3.8/enum.py
def (cls, value)
```

</details>

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,11 @@ matrix:
     - python: 3.9-dev
       env: TOXENV=py39
 
+addons:
+  apt:
+    packages:
+      - libpulse-mainloop-glib0
```
Comment (BryceBeagle) [inline .travis.yml]: What is this needed for?

**Comment by BryceBeagle** [comment]
The signature of `Enum.__new__` matches the runtime signature for each of those enums
https://github.com/python/cpython/blob/3da210b69f76bdd7f4e7fa2e3a12eab4dd478ed2/Lib/enum.py#L574-L580

Does this issue go away if you make these stub classes inherit from `Enum`/`IntEnum` instead of `int`?

File: .travis.yml (hunk 1 above)
Comment (altendky) [reply .travis.yml]: `stubtest` actually processes (runs?) the real package and Qt multimedia stuff had issues.

**Comment by altendky** [comment]
`enum.IntEnum` didn't seem to, but it is possible I was just tired and confused.  For the record, the following issue was mentioned on Gitter as possibly relevant.

https://github.com/python/typeshed/pull/4555#issuecomment-694420214

File: .travis.yml (hunk 1 above)
Comment (BryceBeagle) [reply .travis.yml]:
```suggestion
      # Required to stubtest QtMultimedia
      - libpulse-mainloop-glib0
```

File: stubtest.allowlist.to_review [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+PyQt5.QtNetworkAuth.QAbstractOAuth.Error.__new__
+PyQt5.QtNetworkAuth.QAbstractOAuth.Stage.__new__
+PyQt5.QtNetworkAuth.QAbstractOAuth.Status.__new__
+PyQt5.QtNetworkAuth.QOAuth1.SignatureMethod.__new__
+PyQt5.QtNetworkAuth.QOAuth1Signature.HttpRequestMethod.__new__
+PyQt5.QtNfc.QNearFieldManager.AdapterState.__new__
+PyQt5.QtTest.QAbstractItemModelTester.FailureReportingMode.__new__
+PyQt5.QtWebEngineCore.QWebEngineUrlScheme.Syntax.__new__
+PyQt5.QtWebEngineWidgets.QWebEnginePage.LifecycleState.__new__
+PyQt5.QtWidgets.QActionGroup.ExclusionPolicy.__new__
```
Comment (altendky) [inline stubtest.allowlist.to_review:25]: Regardless we're still waiting on the mypy/stubtest fix to be released...  but aside from that, what do you think of leaving these to be dealt with as part of #106?  Just to get the extra testing in sooner than later (maybe, if mypy gets a release sooner...).

Comment (altendky) [reply stubtest.allowlist.to_review:25]: @BryceBeagle ^

Comment (BryceBeagle) [reply stubtest.allowlist.to_review:25]: Sounds good to me

File: tox.ini [hunk 3]
```diff
@@ -3,12 +3,13 @@ envlist = py3{5,6,7,8,9}
 
 [testenv]
 deps =
-  mypy
+  mypy @ git+https://github.com/python/mypy@538d36481526135c44b90383663eaa177cfc32e3
```
Comment (BryceBeagle) [inline tox.ini:6]: We should probably make an issue to track switching back to a stable release when we can. It looks like this commit will be in v0.800

**Review by BryceBeagle — APPROVED** [review]

File: tox.ini (hunk 3 above)
Comment (altendky) [reply tox.ini:6]: https://github.com/stlehmann/PyQt5-stubs/issues/120

**Review by BryceBeagle — APPROVED** [review]

**Comment by altendky** [comment]
Heh, looks like I should have added GHA with just Linux for now and made Windows/macOS checks separate additions...  `:]`

**Pull request merged**